### PR TITLE
Jump to bootloader from the CLI + Added version field in config for reloading

### PR DESCRIPTION
**Pull request by l-jost**
Bootloader jump sequence:

- Save jump to bootloader in persistent RTC register
- Reset the MCU
- Load the RTC register before system init
- Reset the RTC register
- Jump to bootloader

### CODE REVIEW

File: boards/cats_rev1Pro/src/main.c [hunk 1]
```diff
@@ -84,13 +84,22 @@ int main(void) {
   /* Configure the system clock */
   SystemClock_Config();
 
+  MX_RTC_Init();
+
+  __HAL_RTC_WRITEPROTECTION_DISABLE(&hrtc);
+  HAL_PWR_EnableBkUpAccess();
+  uint32_t d = HAL_RTCEx_BKUPRead(&hrtc, RTC_BKP_DR0);
```
Comment (jbinz) [inline boards/cats_rev1Pro/src/main.c:91]: if(HAL_RTCEx_BKUPRead(&hrtc, RTC_BKP_DR0) == BOOTLOADER_MAGIC_PATTERN)

or rename d to a better name

File: boards/cats_rev1Pro/src/main.h [hunk 2]
```diff
@@ -22,6 +22,8 @@
 /* Includes ------------------------------------------------------------------*/
 #include "stm32l4xx_hal.h"
 
+#define BOOTLOADER_MAGIC_PATTERN 0xAA88BB77
```
Comment (jbinz) [inline boards/cats_rev1Pro/src/main.h:25]: Whats the Magic pattern? Is that the start of the memory for the bootloader?

File: boards/cats_rev1Pro/src/tasks/task_init.c [hunk 3]
```diff
@@ -121,10 +121,11 @@ _Noreturn void task_init(__attribute__((unused)) void *argument) {
 
   osDelay(10);
 
-  /* Check if the FSM configurations make sense */
-  if (global_cats_config.config.control_settings.liftoff_acc_threshold < 1500) {
-    log_error("Acceleration Threshold is not configured Properly!");
```
Comment (jbinz) [inline boards/cats_rev1Pro/src/tasks/task_init.c:126]: are we somehwere making sure that the user cannot set an acceleration threshold lower than 3500? What happens if the user configures the board over USB to have an acceleration threshold of 500?

**Review by jbinz — CHANGES_REQUESTED** [review]

File: boards/cats_rev1Pro/src/cli/cli_commands.c [hunk 4]
```diff
@@ -33,7 +33,7 @@
 #include <string.h>
 
 /** CLI command function declarations **/
-
+static void cli_cmd_bl(const char *cmd_name, char *args);
```
Comment (stojadin2701) [inline boards/cats_rev1Pro/src/cli/cli_commands.c:36]: move this declaration to be together with _reboot() and _save(). They are organized by group

File: boards/cats_rev1Pro/src/cli/cli_commands.c [hunk 5]
```diff
@@ -139,6 +140,14 @@ static void cli_cmd_help(const char *cmd_name, char *args) {
 
 static void cli_cmd_reboot(const char *cmd_name, char *args) { NVIC_SystemReset(); }
 
+extern RTC_HandleTypeDef hrtc;
+static void cli_cmd_bl(const char *cmd_name, char *args) {
+  HAL_RTCEx_BKUPWrite(&hrtc, RTC_BKP_DR0, BOOTLOADER_MAGIC_PATTERN);
+  __disable_irq();
+  NVIC_SystemReset();
+
```
Comment (stojadin2701) [inline boards/cats_rev1Pro/src/cli/cli_commands.c:148]: remove empty line

File: boards/cats_rev1Pro/src/config/cats_config.h [hunk 6]
```diff
@@ -23,6 +23,7 @@
 #include "util/types.h"
 
 /* Exported types */
```
Comment (stojadin2701) [inline boards/cats_rev1Pro/src/config/cats_config.h:25]: remove /* Exported types */

File: boards/cats_rev1Pro/src/config/cats_config.h [hunk 7]
```diff
@@ -23,6 +23,7 @@
...
+#define CONFIG_VERSION 201
```
Comment (stojadin2701) [inline boards/cats_rev1Pro/src/config/cats_config.h:26]: what does 201 represent?

Maybe it's an overkill but maybe we should use semantic versioning?
https://semver.org/

We will use it for sure in the overall code.

If you do this you can for example have:
#define CONFIG_VERSION_MAJOR 2
#define CONFIG_VERSION_MINOR 1

File: boards/cats_rev1Pro/src/config/cats_config.h [hunk 8]
```diff
@@ -39,6 +40,7 @@ typedef enum {
 
 typedef struct {
   /* State according to /concepts/v1/cats_fsm.jpg */
+  uint32_t config_version;
```
Comment (stojadin2701) [inline boards/cats_rev1Pro/src/config/cats_config.h:43]: the comment above this line refers to boot_state, move it down

File: boards/cats_rev1Pro/src/main.c (hunk 1 above)
Comment (stojadin2701) [reply boards/cats_rev1Pro/src/main.c:91]: Please add an explanation here, in 2 months nobody will understand this

File: boards/cats_rev1Pro/src/main.h (hunk 2 above)
Comment (stojadin2701) [reply boards/cats_rev1Pro/src/main.h:25]: Will this pattern change with the version?

File: boards/cats_rev1Pro/src/system_stm32l4xx.c [hunk 9]
```diff
@@ -194,6 +194,44 @@ const uint32_t MSIRangeTable[12] = {100000U,   200000U,   400000U,   800000U,  1
...
+   * Step: Disable systick timer and reset it to default values
+   */
+  SysTick->CTRL = 0;
+  SysTick->LOAD = 0;
+  SysTick->VAL = 0;
+
+  /**
+   * Step: Disable all interrupts
+   */
+  //__disable_irq();
```
Comment (stojadin2701) [inline boards/cats_rev1Pro/src/system_stm32l4xx.c:213]: should this be here if it's commented out?

**Review by stojadin2701 — CHANGES_REQUESTED** [review]
Please add more detail to the commit message, this is also a form of documentation and is more easily accessible than this pull request. If I hover over a line in my editor I can see which commit changed it last and a small description behind the change will help a lot.

Also, mention that you are changing the acc. threshold there.

**Review by jbinz — APPROVED** [review]

**Review by stojadin2701 — APPROVED** [review]

**Pull request merged**